model/riscv_insts_vext_utils.sail

```diff
@@ -0,0 +1,703 @@
+/* ************************************************************************** */
+/* This file implements functions used by vector instructions.                */
+
+/* ************************************************************************** */
+
+
+/* Vector mask mapping */
+mapping maybe_vmask : string <-> bits(1) = {
+  ""              <-> 0b1, /* unmasked by default */
+  sep() ^ "v0.t"  <-> 0b0
+}
+
+/* Check for valid vsew and lmul values */
+val vcheck_vsew_lmul : (int, real) -> Retired
+function vcheck_vsew_lmul(vsew_bits, lmul) = {
+  if  vsew_bits < 8     |
+      vsew_bits > 64    |
+      lmul      < 0.125 |
+      lmul      > 8.0 then {
+    RETIRE_FAIL
+  }
+  else {
+    RETIRE_SUCCESS
+  }
+}
+
+/* Check for vstart value */
+val assert_vstart : int -> Retired effect {rreg}
+function assert_vstart(i) = {
+  if unsigned(readCSR(csr_name_map("vstart"))) != i then {
```

Comment:
Using strings like this is a bad idea, you lose the ability to check typos at compile time. Just use the vstart register directly.

Reply:
CSR reading code is now modified as directly reading.